pkg/internal/ibmblockcsi/default_setter.go

```diff
@@ -18,31 +18,107 @@ package ibmblockcsi
 
 import (
 	"github.com/IBM/ibm-block-csi-operator/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+	"path"
 )
 
 // SetDefaults set defaults if omitted in spec, returns true means CR should be updated on cluster.
 // Replace it with kubernetes native default setter when it is available.
 // https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#defaulting
 func (c *IBMBlockCSI) SetDefaults() bool {
-	// repository is mandatory, tag is optional, but if repository is not set
-	// and tag is set, the tag will be overrided to the default one.
-	changed := false
+	if c.isAnyNonOfficialRepo() {
```

Comment:
Let's change `NonOfficial` to `Unofficial` as in:
```suggestion
	if c.isAnyUnofficialRepo() {
```
(all over)

Reply:
done